### PR TITLE
Update ios-device-identifiers.json

### PR DESCRIPTION
**Pull request by pschmidtboeing**
Added missing 8th Gen iPads, 4th Gen iPad Airs, and iPhones 12 models. Updated the sort/grouping to align with their introduction dates (oldest to newest).

I also updated the names iPad8,9 and iPad8,10 to contain "4th Gen" because they (along with the iPad8,11 and iPad8,12) are the "4th Generation" iPad Pros that were all introduced on March 18, 2020. Although other sites have labelled the iPad8,9 with "iPad Pro 2nd Gen (11 inch, WiFi)" and iPad8,10 with "iPad Pro 2nd Gen (11 inch, WiFi+Cellular)", I believe those names are misleading and will likely cause people to incorrectly think they are 2 model years older than they actually are. It seems "2nd Gen" was chosen because they are the 2nd generation of the 11" iPad, but this reasoning doesn't align with the naming convention of other products with different screen sizes. For example: iPad11,6 and iPad11,7 are "8th Gen" even though they have 10.2" screens and all prior generations had 9.7" screens.

### CODE REVIEW

File: ios-device-identifiers.json [hunk 1]
```diff
@@ -25,19 +25,23 @@
...
+  "iPhone10,2": "iPhone 8 Plus",
   "iPhone10,5": "iPhone 8 Plus (no CDMA)",
+  "iPhone10,3": "iPhone X",
   "iPhone10,6": "iPhone X (no CDMA)",
-  "iPhone11,2": "iPhone XS",
-  "iPhone11,4": "iPhone XS Max (special model?)",
-  "iPhone11,6": "iPhone XS Max",
+  "iPhone11,2": "iPhone Xs",
+  "iPhone11,4": "iPhone Xs Max (special model?)",
+  "iPhone11,6": "iPhone Xs Max",
```
Comment (timdiggins) [inline ios-device-identifiers.json:35]: interesting -- Apple seem to be unsure themselves how to show the `s` in iPhone XS/Xs. I think maybe in fact like the iPhone XR it's meant to be a small-caps upper case, but that doesn't make life easy in Ascii. 

Refurbished_iPhone_XS_64GB_-_Silver__Unlocked__-_Apple
iPhone_Xs_and_iPhone_Xs_Max_bring_the_best_and_biggest_displays_to_iPhone_-_Apple__UK_
iPhone_XS_vs_iPhone_12_vs_iPhone_12_mini_-_Apple__UK_
iPhone XS_Max_vs_iPhone XR_vs_iPhone 12 mini_-_Apple

I'm willing to accept lower case!

**Comment by timdiggins** [comment]
@pschmidtboeing Thanks for your PR and thanks for letting me know the logic of the ordering.

File: ios-device-identifiers.json (hunk 1 above)
Comment (pschmidtboeing) [reply ios-device-identifiers.json:35]: Based on the image, I agree with you that it looks like the S in "XS" is a smaller, yet still uppercase S, much like the R in "XR".

Comment (timdiggins) [reply ios-device-identifiers.json:35]: Agreed. I've changed it to XS with caps see #9

**Pull request merged**